### PR TITLE
Update docs on how to record removed API

### PR DESCRIPTION
**Pull request by RussKie**

### CODE REVIEW

File: src/PublicApiAnalyzers/Core/Analyzers/PublicApiAnalyzerResources.resx [hunk 1]
```diff
@@ -160,7 +160,7 @@
     <value>Symbol '{0}' is part of the declared API, but is either not public or could not be found</value>
   </data>
   <data name="RemoveDeletedApiDescription" xml:space="preserve">
-    <value>When removing a public type or member the corresponding entry in PublicAPI.txt should also be removed. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.</value>
+    <value>When removing a public type or member put that entry in PublicAPI.Unshipped.txt with *REMOVED* prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.</value>
```
Comment (Youssef1313) [inline src/PublicApiAnalyzers/Core/Analyzers/PublicApiAnalyzerResources.resx:163]: Also it's safer to lock `*REMOVED*` from localization. Syntax for locking localization:

https://github.com/dotnet/roslyn-analyzers/blob/affa88c580a9cffc5b016ad892c5bcd3971e8456/src/Roslyn.Diagnostics.Analyzers/Core/RoslynDiagnosticsAnalyzersResources.resx#L353

```suggestion
    <value>When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '*REMOVED*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.</value>
```

File: src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ All public types and members should be declared in PublicAPI.txt. This draws att
 
 ## [RS0017](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md): Remove deleted types and members from the declared API
 
-When removing a public type or member the corresponding entry in PublicAPI.txt should also be removed. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.
+When removing a public type or member put that entry in PublicAPI.Unshipped.txt with *REMOVED* prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.
```
Comment (Youssef1313) [inline src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md:17]: 🐛 This is a bug in markdown generation. The asterisks should be escaped. I'll file a follow-up issue on that.

Comment (RussKie) [reply src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md:17]: Good catch, it didn't occur to me.

**Review by drewnoakes — COMMENTED** [review]
Is the REMOVED string a symbol that should not be translated?

**Comment by Youssef1313** [comment]
> Is the REMOVED string a symbol that should not be translated?

Yeah I think it shouldn't. I requested to use "Locked" in https://github.com/dotnet/roslyn-analyzers/pull/5210#discussion_r665960432, but I think this part of the commend was unintentionally dropped.

**Comment by RussKie** [comment]
> I requested to use "Locked" in [#5210 (comment)](https://github.com/dotnet/roslyn-analyzers/pull/5210#discussion_r665960432), but I think this part of the commend was unintentionally dropped.

Totally missed it! Thank you for pointing it out.

**Comment by Youssef1313** [comment]
@RussKie No nees to manually update doc files. You can run `msbuild /t:pack`, they will get updates automatically.

**Comment by RussKie** [comment]
I have... It does something very bizarre like replacing the string I modified with a completely different string. I'll try again later...

File: src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md [hunk 3]
```diff
@@ -14,7 +14,7 @@ All public types and members should be declared in PublicAPI.txt. This draws att
 
 ## [RS0017](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md): Remove deleted types and members from the declared API
 
-When removing a public type or member the corresponding entry in PublicAPI.txt should also be removed. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.
+When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '\*REMOVED\*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.
```
Comment (Youssef1313) [inline src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md:17]: This suggestion removes the "correct" escaping. This should be fixed in the tool generating the doc file.

```suggestion
When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '*REMOVED*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.
```

File: src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.sarif [hunk 4]
```diff
@@ -31,7 +31,7 @@
         "RS0017": {
           "id": "RS0017",
           "shortDescription": "Remove deleted types and members from the declared API",
-          "fullDescription": "When removing a public type or member the corresponding entry in PublicAPI.txt should also be removed. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.",
+          "fullDescription": "When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '\*REMOVED\*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.",
```
Comment (Youssef1313) [inline src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.sarif:34]:
```suggestion
          "fullDescription": "When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '*REMOVED*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.",
```

File: src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md (hunk 3 above)
Comment (RussKie) [reply src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md:17]: Thank you.

**Comment by Youssef1313** [comment]
> I have... It's doing something very bizarre like replacing the string I modified with a completely different string. I'll try again later...

If it's about the escaping backslash, then it's okay, fixing the tool is tracked by #5219.
If it's doing some other weird thing, it could be a another unknown bug with the tool.

**Comment by RussKie** [comment]
I have rebased my branch on top of the latest main, `git clean -xdf`, followed by `build.cmd -pack`. As a result I get this changes:
image

This makes no sense to me...

**Comment by Youssef1313** [comment]
@RussKie I can reproduce. For some reason, the description of the diagnostic descriptor is empty:

image

I don't really get why :(

**Comment by Youssef1313** [comment]
The weird thing is that this problem doesn't exist in CI (ie, the CI is failing and is most likely expecting the correct result, at the same time, both CI and local development relies on the same generation code)

**Comment by Youssef1313** [comment]
I decompiled the produced artifacts using Telerik JustDecompile. Here is the resources I see in the decompiler:

image

The dll I decompiled is located at `<repo-root>\artifacts\bin\Microsoft.CodeAnalysis.PublicApiAnalyzers\Debug\netstandard1.3\Microsoft.CodeAnalysis.PublicApiAnalyzers.dll`

I have no idea how to dig further. But I hope you find this information useful.

File: src/PublicApiAnalyzers/Core/Analyzers/PublicApiAnalyzerResources.resx [hunk 5]
```diff
@@ -160,7 +160,8 @@
     <value>Symbol '{0}' is part of the declared API, but is either not public or could not be found</value>
   </data>
   <data name="RemoveDeletedApiDescription" xml:space="preserve">
-    <value>When removing a public type or member the corresponding entry in PublicAPI.txt should also be removed. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.</value>
+    <value>When removing a public type or member, put that entry in PublicAPI.Unshipped.txt with '*REMOVED*' prefix. This draws attention to API changes in the code reviews and source control history, and helps prevent breaking changes.</value>
+    <note>{Locked="*REMOVED*"}</note>
```
Comment (Youssef1313) [inline src/PublicApiAnalyzers/Core/Analyzers/PublicApiAnalyzerResources.resx:164]: Ohhh, this should have been `<comment>` not `<note>`

I haven't confirmed if that fixes the generation, but it should, I can't imagine there is any other issue.

Comment (RussKie) [reply src/PublicApiAnalyzers/Core/Analyzers/PublicApiAnalyzerResources.resx]: 🤦‍♂️ I looked at a wrong file.... in xlf files the "comment" is turned into a "note"

**Comment by RussKie** [comment]
Yay! It is finally green!

Thank you @Youssef1313 so much for your help!

**Review by jmarolf — APPROVED** [review]

**Pull request merged**